pkg/cli/admin/release/extract_tools.go

```diff
@@ -490,7 +492,7 @@ func (o *ExtractOptions) extractCommand(command string) error {
 		// copy the input to disk
 		if target.InjectReleaseImage {
 			var matched bool
-			matched, err = copyAndReplaceReleaseImage(w, r, 4*1024, exactReleaseImage)
+			matched, err = copyAndReplaceReleaseImage(w, r, 4*1024, releaseName)
```

Comment:
I believe we want to report releaseName here instead of exactReleaseImage? sha is reported in sha256sum.txt 
For example, `openshift-install version`, output is 
`release image 4.2.0-0.ci-2019-09-09-042520` instead of
`release image registry.svc.ci.openshift.org/ocp/release@sha256:371460b3801ad48b58a6456464.......`

Reply:
> For example, openshift-install version, output is
> release image 4.2.0-0.ci-2019-09-09-042520 instead of
> release image registry.svc.ci.openshift.org/ocp/release@sha256:371460b3801ad48b58a6456464.......

We want the by-digest release pullspec in the installer, because then a signed installer is secure against attacks on the mutable tag.  The short version name is nice to display as well, but we want both the short version name and the by-digest pullspec in the installer, not just the short version name.  And the short version name doesn't have to be part of a pullspec.  You could push to registry.example.com/what/ever:you-want, and you'd still want to inject the `version` field in the image's `release-metadata` as the version string, not whatever the pullspec tag was.

Reply:
and I see now when I try to create cluster the installer needs the by-digest. : )

Reply:
Yes, but I don't think `oc` itself needs that by-digest. For oc only the short is sufficient.

Reply:
So you'll have to have a one for oc and a different for installer.

Reply:
right, I'll simply do an if on the targetName :) got it

Reply:
I think the installer wants *both* the version and by-digest pullspec.  So either sub in for two placeholders, or sub something structured into a single placeholder.

Reply:
that's what I was thinking.. it'll require a change in installer code (i have that worked out already), so installer wants to print short form for version cmd? then needs pullspec to install, right?

Reply:
> ... installer wants to print short form for version cmd? then needs pullspec to install, right?

Yup.  And it actually prints both for the `version` command.